datafusion/core/tests/fuzz_cases/aggregation_fuzzer/fuzzer.rs

```diff
@@ -0,0 +1,266 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::sync::Arc;
+
+use arrow::util::pretty::pretty_format_batches;
+use arrow_array::RecordBatch;
+use tokio::task::JoinSet;
+
+use crate::fuzz_cases::aggregation_fuzzer::{
+    check_equality_of_batches,
+    context_generator::{SessionContextGenerator, SessionContextWithParams},
+    data_generator::{Dataset, DatasetGenerator, DatasetGeneratorConfig},
+    run_sql,
+};
+
+/// Aggregation fuzzer's builder
+pub struct AggregationFuzzerBuilder {
+    /// See `data_gen_rounds` in [`AggregationFuzzer`], default 50
+    data_gen_rounds: usize,
+
+    /// See `ctx_gen_rounds` in [`AggregationFuzzer`], default 50
+    ctx_gen_rounds: usize,
+
+    /// See `sql` in [`AggregationFuzzer`], no default, and required to set
+    sql: Option<Arc<str>>,
+
+    /// See `table_name` in [`AggregationFuzzer`], no default, and required to set
+    table_name: Option<Arc<str>>,
+
+    /// Used to generate `dataset_generator` in [`AggregationFuzzer`],
+    /// no default, and required to set
+    data_gen_config: Option<DatasetGeneratorConfig>,
+}
+
+impl AggregationFuzzerBuilder {
+    fn new() -> Self {
+        Self {
+            data_gen_rounds: 50,
+            ctx_gen_rounds: 50,
```

Comment:
I wonder if we really need 50 each time, or if we can reduce this to 10 each or something and still keep coverage

Reply:
Indeed too large as a default value. 
I overwrite the `data_gen_rounds` to 20 when I write actual tests, but seems still too slow.
Maybe 20 * 20 is better?

Reply:
Is it possible to make sure each combination is tested, then the actual task number is `ctx_gen_rounds` * 8.

Reply:
> Is it possible to make sure each combination is tested, then the actual task number is `ctx_gen_rounds` * 8.

🤔 `ctx_gen_rounds` represent that, the `generate` of `SessionContextGenerator` will be called `ctx_gen_roudns` times.

And each flag (spilling or not, partial skipping or not, intting sorted or not...) will get a random value (true/false) in each `generate` calling.

So seems 8(2 * 2 * 2) will be possible to make sure all mode combination can be tested for a single dataset(will generate `ctx_gen_rounds` session contexts for each dataset).

And the total `dataset num` will be `data_gen_rounds * len(sort_keys_set)`, so the `task_num` will be `data_gen_rounds * len(sort_keys_set) * ctx_gen_rounds`.